pkg/helm/environment/environment.go

```diff
@@ -32,8 +34,21 @@ import (
 	"k8s.io/helm/pkg/helm/helmpath"
 )
 
-// defaultHelmHome is the default HELM_HOME.
-var defaultHelmHome = filepath.Join(homedir.HomeDir(), ".helm")
+var oldDefaultHelmHome = filepath.Join(homedir.HomeDir(), ".helm")
+var defaultHelmHome = filepath.Join(xdg.ConfigHome(), "helm")
+
+// Get configuration home dir.
+//
+// Note: Temporal until all migrate to XDG Base Directory spec
+func getDefaultConfigHome() string {
+	if _, err := os.Stat(defaultHelmHome); err != nil {
+		return defaultHelmHome
+	} else if _, err := os.Stat(oldDefaultHelmHome); err != nil {
+		return defaultHelmHome
+	}
+	fmt.Printf("WARNING: using old-style configuration directiry. Please, consider moving it to %s\n", defaultHelmHome)
```

Comment:
It might be better if this warning was moved to the `cmd` package anyways, as calling `fmt.Fprintf` here doesn't allow us to print it on different outputs (to a file, stderr, etc).